### DISCUSSION TITLE
interface implementations types do not inherit its data fetchers?

### DISCUSSION THREAD

**Discussion by gust-gb** [graphql-java/graphql-java#2444] [Q&A] [answered]
sorry if I misunderstanding interface mechanism but I thought that its implementations would inherit its data fetchers, so the problem for me is:

I have an interface called `Account` which is implemented by `CheckingAccount` and `CreditCardAccount`

`Account` interface has an attribute like `transactions: [Transaction]` and I created a data fetcher for it

I would expected that every `CheckingAccount` and `CreditCardAccount` could resolve `transactions` too, but this is not happening, it's returning null (because, as I see, there is no `transactions` data fetcher defined for them, just for interface)

is this by design? could I configure something for this?

I don't know if it is the same problem, but one guy point that out for GraphQL Tools on JS and there is a configuration you can inject building schema (`inheritResolversFromInterfaces: true`): https://stackoverflow.com/questions/50848150/graphql-object-that-implements-an-interface-is-not-not-inheriting-resolver-from

so I tried to put some options in `SchemaGenerator.makeExecutableSchema` but `Options` class does not have any attributes and `makeExecutableSchema` method which accepts `Options` parameter just ignore it

I'm appologize if I'm making confusion and would be great if anyone could help me!

**Comment by bbakerman** [comment, marked as answer]
> I would expected that every CheckingAccount and CreditCardAccount could resolve transactions too, but this is not happening, it's returning null (because, as I see, there is no transactions data fetcher defined for them, just for interface)

> is this by design? could I configure something for this?

Its is by design - the runtime engine only ever deals with concrete object types.

When an interface field is used

```
{
   account(id :123434) {
       transactions { 
         amount
      }
 }
```

The engine calls a `graphql.schema.TypeResolver` to decide on what GraphQL object type maps to the Java object in memory.

once it has that, it looks up the data fetcher from object type by name.  This allows (and requires) you to have specific data fetchers for specific fields in specific types.

if you are looking for a more system way to wire in data fetchers, check out `graphql.schema.idl.WiringFactory#providesDataFetcher` and `graphql.schema.idl.WiringFactory#getDataFetcher`.  This is a interceptor like way to decide what fields get what data fetchers

**Reply by gust-gb** [reply]
thank you!
